developer/src/kmc-ldml/src/compiler/tran.ts

```diff
@@ -199,6 +191,33 @@ export abstract class TransformCompiler<T extends TransformCompilerType, TranBas
     return result;
   }
 
+  /**
+   * Validate the final regex
+   * @param cookedFrom the regex to use, missing the trailing '$'
+   * @param from the original from - for error reporting
+   * @returns true if OK
+   */
+  private isValidRegex(cookedFrom: string, from: string) : boolean {
+    // check for any unescaped dollar sign here
+    if (/(?<!\\)\$/.test(cookedFrom)) {
+      this.callbacks.reportMessage(CompilerMessages.Error_UnparseableTransformFrom({ from, message: `Transform contains '$' but did not match any variable substitution. '$' is not valid in transform syntax.` }));
+      return false;
+    }
```

Comment:
The error string here needs to be in the CompilerMessages object for future translation, and so we can add mitigations and resolution text later. So we'll need to give it its own error id.

Reply:
This regex does not match on `\\$` which is also invalid (i.e. `\` is escaped but `$` is not).

Reply:
- will give it its own number. Perhaps a number-sub-space since there will be a few of these kinds of messages?
- will fix the regex to match if there is an _even_ number of backslashes.

Reply:
nice! or dreadful, one or the other. Can never quite decide when it comes to regex